config_files/training/config_lorem_ipsum_sft.yaml

```diff
@@ -0,0 +1,334 @@
+settings:  
+  experiment_id: ${modalities_env:experiment_id}
+  config_file_path: ${modalities_env:config_file_path}
+  referencing_keys:
+    sample_key: input_ids
+    target_key: target_ids
+  training:
+    training_log_interval_in_steps: 2
+    checkpointing_interval_in_steps: 4
+    evaluation_interval_in_steps: 2
+    global_num_seen_tokens: 0
+    activation_checkpointing_modules: [GPT2Block]
+    gradient_acc_steps: 2
+    local_train_micro_batch_size: 1
+    sequence_length: 2048
+  cuda_env:
+    local_rank: ${cuda_env:LOCAL_RANK}
+    global_rank: ${cuda_env:RANK}
+    world_size: ${cuda_env:WORLD_SIZE}
+  paths:
+    checkpointing_path: data/checkpoints
+
+collate_fn:  
+  component_key: collate_fn
+  variant_key: mask_loss_collator_wrapper
+  config:
+    collate_fn:  
+      component_key: collate_fn
+      variant_key: gpt_2_llm_collator
+      config:
+        sample_key: ${settings.referencing_keys.sample_key}
+        target_key: ${settings.referencing_keys.target_key}
+    target_keys_to_mask:
+      - ${settings.referencing_keys.target_key}
+    loss_ignore_index: -100
+    mask_tokens:
+      b_include_to_loss_token: ^
+      e_include_to_loss_token: $
```

Comment:
That is a good point. Ideally, we should have special tokens guaranteed not to be part of the content of the SFT data. Sadly, this is very tokenizer-specific. For example, if we want to train an HF model, it is very custom what tokens are there, e.g., if there are reserved placeholder tokens. As this token handling is very much work, a separate PR would be better. See feature request #208

Reply:
For now, the user must make sure this is a single token. During loss masking, we check if the sequence makes sense, e.g., end and start tokens are alternating.

It could still happen that "^" and "$" are in this order in the sequence. This can only be resolved after using dedicated special tokens. See #208

Reply:
> For now, the user must make sure this is a single token. During loss masking, we check if the sequence makes sense, e.g., end and start tokens are alternating.

Do we throw an error if there is an input string with ambiguous place holders?

Reply:
Yes, when we run the training, we check if the special tokens are new tokens unknown to the vocab; if so, we throw an error, as we don't support resizing the embedding mask, see #208 

We also check that the tokens for loss masking, which are now marked as special, alternate during loss masking.